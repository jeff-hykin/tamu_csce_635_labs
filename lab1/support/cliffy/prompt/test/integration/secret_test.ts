import { Secret } from "../../secret.ts";
import { snapshotTest } from "../../../testing/snapshot.ts";

await snapshotTest({
  name: "secret prompt",
  meta: import.meta,
  osSuffix: ["windows"],
  steps: {
    "should enter a secret": { stdin: ["123", "\n"] },
  },
  async fn() {
    await Secret.prompt({
      message: "enter your secret",
    });
  },
});
