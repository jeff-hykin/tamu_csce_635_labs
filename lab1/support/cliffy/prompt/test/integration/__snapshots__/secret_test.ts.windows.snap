export const snapshot = {};

snapshot[`secret prompt > should enter a secret 1`] = `
stdout:
"? enter your secret » \\x1b[23G\\x1b[G\\x1b[0J
? enter your secret » ***\\x1b[26G\\x1b[G\\x1b[0J
? enter your secret » ***
\\x1b[?25h\\x1b[?25h"
stderr:
""
`;
